python/pyspark/sql/connect/group.py

```diff
@@ -414,6 +414,58 @@ def transformWithStateInPandas(
 
     transformWithStateInPandas.__doc__ = PySparkGroupedData.transformWithStateInPandas.__doc__
 
+    def transformWithState(
+        self,
+        statefulProcessor: StatefulProcessor,
+        outputStructType: Union[StructType, str],
+        outputMode: str,
+        timeMode: str,
+        initialState: Optional["GroupedData"] = None,
+        eventTimeColumnName: str = "",
+    ) -> "DataFrame":
+        from pyspark.sql.connect.udf import UserDefinedFunction
+        from pyspark.sql.connect.dataframe import DataFrame
+        from pyspark.sql.streaming.stateful_processor_util import (
+            TransformWithStateInPySparkUdfUtils,
+        )
+
+        udf_util = TransformWithStateInPySparkUdfUtils(statefulProcessor, timeMode)
+        if initialState is None:
+            udf_obj = UserDefinedFunction(
+                udf_util.transformWithStateUDF,
+                returnType=outputStructType,
+                evalType=PythonEvalType.SQL_TRANSFORM_WITH_STATE_UDF,
```

Comment:
@HeartSaVioR Did some code changes perhaps fail to be submitted ? After the merge, both the testing and lint checks failed.

- https://github.com/apache/spark/actions/runs/14689575181/job/41222900672

![image](https://github.com/user-attachments/assets/2c504275-1c77-4d76-bfc5-4fe31cea5ae3)

- https://github.com/apache/spark/actions/runs/14689575181/job/41222899818

![image](https://github.com/user-attachments/assets/f964ff57-bb70-468a-8c32-a313b5599247)

Reply:
Try to fix: https://github.com/apache/spark/pull/50734